File: solutions/ocaml/150-design-hit-counter.ml
(*
  Problem 150: Design Hit Counter (LeetCode 362)
  Difficulty: Med
  Language: OCaml
*)
type hit_counter = int list

let hit hc ts = hc @ [ts]

let get_hits hc ts =
  let filtered = List.filter (fun t -> t > ts - 300) hc in
  (List.length filtered, filtered)

File: solutions/haskell/150-design-hit-counter.hs
{-
  Problem 150: Design Hit Counter (LeetCode 362)
  Difficulty: Med
  Language: Haskell
-}
type HitCounter = [Int]

hit :: HitCounter -> Int -> HitCounter
hit hc ts = hc ++ [ts]

getHits :: HitCounter -> Int -> (Int, HitCounter)
getHits hc ts =
  let filtered = filter (> ts - 300) hc
  in (length filtered, filtered)
